sort_names.py

```diff
@@ -1,154 +1,174 @@
+        "count_name",
+        "count_last",
+        "count_pos",
+    ]
+    for cond, func in zip(conditions1, functions1):
+        if cond:
+            iterate_count(func)
+    if len(counts) < 3:
+        iterate_count("count_simple")
+        return counts
+    conditions2 = [
+        (row[30] == "") and DEL_PEOPLE_WITHOUT_AVATAR,
+        (row[30] == "") and not DEL_PEOPLE_WITHOUT_AVATAR,
+    ]
+    functions2 = [
+        "count_no_link",
+        ("count", "count_no_link"),
+    ]
+    for cond, func in zip(conditions2, functions2):
+        if cond:
+            iterate_count(func)
+    if counts["count_no_link"]:
+        return counts
+    filter_face()
+    return counts
+
+
+def counters(dict_, dict_child):
+    """Find goal filter in dict_child and rewrite (increments) some values in dict_"""
+    for elem in dict_child.keys():
+        if dict_child[elem]:
+            dict_[elem] += 1
+
+
+def get_short_list(f_name, first_names, last_names, positions):
     """Read .csv file and create(rewrite) another .csv file with short list of people"""
-    count = 0
+    d_counters = {
+        "count": 0,
+        "count_no_link": 0,
+        "count_unreadable_ava": 0,
+        "count_name": 0,
+        "count_last": 0,
+        "count_pos": 0,
+        "count_age": 0,
+        "count_race": 0,
+        "count_simple": 0,
+    }
     count_row = 0
-    count_no_link = 0
     n_cond = 0
-    res_file_name = filename[:-4] + "_filt.csv"
-    res_bad_file_name = filename[:-4] + "_without_avatar.csv"
-    with open(filename, "r", encoding="utf8") as file, open(
-        res_file_name, "w+", encoding="utf8", newline=""
-    ) as res_file, open(
-        res_bad_file_name, "w+", encoding="utf8", newline=""
-    ) as res_bad_file:
+    d_rases = defaultdict(int)
+    res_file_name = f_name[:-4] + "_filt.csv"
+    res_del_file_name = f_name[:-4] + "_deleted.csv"
+    with open(res_file_name, "w+", encoding="utf8", newline="") as res_file, open(
+        res_del_file_name, "w+", encoding="utf8", newline=""
+    ) as res_del_file:
         writer = csv.writer(res_file, delimiter=";")
-        writer2 = csv.writer(res_bad_file, delimiter=";")
-        csvreader = csv.reader(
-            file, delimiter=","
-        )  # delimiter="," - if the data is concatenated in the first column by ','
-        # delimiter=";" - if the data in the different column
-        first_row = next(csvreader)
-        read_file = list(csvreader)
+        writer2 = csv.writer(res_del_file, delimiter=";")
+        read_file, first_row = inputs.read_file(f_name)
         num_people = len(read_file)
-        count_name = count_last = count_pos = count_age_race = num_people
-        print("--- START ---")
-        print("Estimated time ~ {} minutes".format(round(num_people * 1.2 / 60)))
+        print(f"Estimated time ~ {round(num_people * 1.2 / 60)} minutes")
         writer.writerow(first_row)
         writer2.writerow(first_row)
 
         for row in read_file:
```

Comment:
can we extract this and next block into functions and pass `writers` as parameters?